helix-rest/src/main/java/org/apache/helix/rest/server/resources/lock/LockAccessor.java

```diff
@@ -0,0 +1,228 @@
+    } catch (Exception e) {
+      return badRequest(e.getMessage());
+    }
+
+    // Perform try lock
+    boolean result;
+    try {
+      result = lock.tryLock();
+    } catch (HelixException e) {
+      return badRequest(
+          "Lock path is not found or lock scope could not be generated, please examine the request.");
+    }
+
+    return result ? OK(JSONRepresentation(lock)) : serverError(String.format(
+        "Failed to lock the path %s using zk address %s, please check if this lock is held by someone else.",
+        lockLocator.getLockPath(), lockLocator.getZkAddress()));
+  }
+
+  private LockLocator parseLockLocator(ZNRecord znRecord) throws Exception {
+    if (znRecord == null) {
+      return null;
+    }
+
+    // Parse zk address from simple field
+    String zkAddr = znRecord.getSimpleField(ZK_ADDR);
+    if (zkAddr == null) {
+      throw new Exception("No ZK address is provided, could not create a lock.");
+    }
+
+    // Parse lock path from simple field
+    String lockPath = znRecord.getSimpleField(LOCK_PATH);
+
+    // If lock path is not provided, generate a lock path from lock scope
+    if (lockPath == null) {
+      Map<String, Map<String, String>> mapFields = znRecord.getMapFields();
+      LockScope lockScope = null;
+      for (LockScopeKey lockScopeKey : LockScopeKey.values()) {
+        // If a known lock scope is found, currently only HelixLockScope is available
+        if (mapFields.containsKey(lockScopeKey.name())) {
+          // Helix Lock Scope
+          if (lockScopeKey.name().equals(LockScopeKey.HelixLockScope.name())) {
+            lockScope = generateHelixLockScope(mapFields.get(lockScopeKey.name()));
+          }
+        }
+      }
+      if (lockScope == null) {
+        throw new Exception(
+            "Failed to parse a lock scope. Please check the lock scope property and path keys and make sure they are correct.");
+      }
+      return new LockLocator(zkAddr, lockScope.getPath());
+    }
+
+    return new LockLocator(zkAddr, lockPath);
+  }
+
+  private LockInfo parseLockInfo(ZNRecord znRecord) throws Exception {
+    if (znRecord == null) {
+      return null;
+    }
+    // Parse user iod
+    String userId = znRecord.getSimpleField(USER_ID);
+
+    // Parse lock timeout
+    long lockTimeout;
+    try {
+      lockTimeout = Long.parseLong(znRecord.getSimpleField(LOCK_TIMEOUT));
+    } catch (NumberFormatException e) {
+      throw new Exception("Please check the timeout value input, could not convert to long.");
+    }
+
+    // Parse lock message
+    String lockMsg = znRecord.getSimpleField(LOCK_MSG);
+    return new LockInfo(userId, lockMsg, lockTimeout);
+  }
+
+  private LockScope generateHelixLockScope(Map<String, String> lockScopeVal) throws Exception {
+    // Get the lock scope property: cluster, participant, resource, etc.
+    String CLUSTER = "cluster";
+    String RESOURCE = "resource";
+    String PARTICIPANT = "participant";
```

Comment:
It looks like we already have the enum for it. why can't we just use the enum instead of defining new static constants?

Either add a method in the enum or just do a enum.getString().toLowerCase().